File: 71.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        arr = path.split("/")
        stack = []
        
        for cur in arr:
            if cur == "..":
                if stack:
                    stack.pop()
            elif cur != "." and cur != "":
                stack.append(cur)
                
        return "/" + "/".join(stack)

File: 71.java
class Solution {
    public String simplifyPath(String path) {
        String[] arr = path.split("/");
        Stack<String> stack = new Stack<>();
        
        for (String cur : arr) {
            if (cur.equals("..")) {
                if (!stack.empty()) {
                    stack.pop();
                }
            } else if (!cur.equals(".") && !cur.equals("")) {
                stack.push(cur);
            }
        }
        return "/" + String.join("/", stack);
    }
}